cmd/monitor/monitor.go

```diff
@@ -0,0 +1,201 @@
+func validateAddress(ipPort string) (string, error) {
+	ipPort = whitespaceFilter.ReplaceAllString(ipPort, "")
+	pts := strings.Split(ipPort, ":")
+	if len(pts) != 2 {
+		return "", ErrInvalidAddress
+	}
+	ip := net.ParseIP(pts[0])
+	if ip == nil {
+		return "", ErrInvalidAddress
+	} else if ip.IsLoopback() {
+		if false {
+			return "", ErrNoLocalhost
+		}
+	} else if !ip.IsGlobalUnicast() {
+		return "", ErrNotExternalIP
+	}
+
+	port, err := strconv.ParseUint(pts[1], 10, 16)
+	if err != nil {
+		return "", ErrInvalidAddress
+	}
+
+	if port < 1024 {
+		return "", ErrPortTooLow
+	}
+
+	return ipPort, nil
+}
+
+// parseRemotePeerList parses a remote peers.txt file
+// The peers list format is newline separated ip:port
+// Any lines that don't parse to an ip:port are skipped
+// Localhost ip:port addresses are ignored
+func parseRemotePeerList(body string) []string {
+	var peers []string
+	for _, addr := range strings.Split(string(body), "\n") {
+		addr = whitespaceFilter.ReplaceAllString(addr, "")
+		if addr == "" {
+			continue
+		}
+
+		// Never allow localhost addresses from the remote peers list
+		a, err := validateAddress(addr)
+		if err != nil {
+			logger.Errorf("Remote peers list has invalid address %s: %v", addr, err)
+			continue
+		}
+
+		peers = append(peers, a)
+	}
+
+	return peers
+}
+
+type peer struct {
+	Addr            string `json:"address"`
+	Trusted         bool   `json:"trusted"`
+	StartConnection string `json:"timeStartConnection"`
+	GetConnection   string `json:"timeGetConnection"`
+}
+
+func getPeers() ([]string, map[string]peer) {
+	file, err := os.Open(urlPeers)
+	if err != nil {
+		logger.WithError(err).Error(file)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		logger.WithError(err).Error(b)
+		os.Exit(1)
+	}
+
+	pexList := parseRemotePeerList(string(b))
+
+	var output []string
+	peers := make(map[string]peer, len(pexList))
+	for _, pex := range pexList {
+		conn, err := net.DialTimeout("tcp", pex, dialTimeout)
+		peers[pex] = peer{
+			Addr:            pex,
+			Trusted:         false,
+			StartConnection: "nil",
+			GetConnection:   "nil",
+		}
+		if err != nil {
+			logger.WithError(err).Errorf("net.DialTimeout")
+			continue
+		}
+		defer conn.Close()
+		conn.Write([]byte("GET / HTTP/1.0\r\n\r\n"))
+
+		start := time.Now()
+		oneByte := make([]byte, 1)
+		_, err = conn.Read(oneByte)
+		if err != nil {
+			logger.WithError(err).Errorf("Read")
+			continue
+		}
+		timeStart := time.Since(start)
+
+		_, err = ioutil.ReadAll(conn)
+		if err != nil {
+			logger.WithError(err).Errorf("ReadAll")
+			continue
+		}
+		timeEnd := time.Since(start)
+
+		output = append(output, pex)
+		bufferpex := peer{Addr: pex, Trusted: true, StartConnection: timeStart.String(), GetConnection: timeEnd.String()}
+		peers[pex] = bufferpex
+
+	}
+
+	return output, peers
+
+}
+func main() {
+
+	urlPeer := flag.String("f", urlPeers, "Url the file peer.txt")
```

Comment:
You should have a configuration struct with fields initially set to default values and populated by `flag` afterwards. Parsing should happen inside a method (receiver) named `ParseConfig` .

Reply:
Thats not necessary for a simple cli

Reply:
👍

Reply:
@gz-c I'll approve changes then . Please review .

Reply:
Oh ! Noticed something ...